primal-sieve/src/sieve.rs

```diff
@@ -253,21 +253,19 @@ impl Sieve {
 
         let mut ret = Vec::new();
 
-        for p in self.primes_from(0) {
-            let mut count = 0;
-            while n % p == 0 {
+        self.primes_from(0).all(|p| {
```

Comment:
I personally find the use of `all` for a purely-side-effecting loop like this a bit confusing, but I guess it's linking into the `all` specialisation? Another option that directly expresses the goal would be `try_for_each`, I think? (It seems this was stabilised in 1.27?)

Reply:
Yeah, basically we want specialized internal iteration with early termination (`while p*p < n`), which is perfect for `try_fold` or `try_for_each`, except we can't actually specialize those because the `Try` trait itself is still unstable.

This is local code though, so maybe I should invent an inherent method that's clearer, instead of abusing `all`...

Reply:
Probably not worth it for a single call site, vs. just adding a brief comment like `// use internal iteration` or something?

Reply:
I realized that my `all` was incorrect, because it didn't bother updating `self.early`, and iterators are allowed to be called again afterward. So I decided to customize it, `for_each_while`.